### PR TITLE
Enable ESLint `prefer-rest-params` rule

### PR DESCRIPTION
**Pull request by fisker**
Notice: Similar changes inside `fast-path.js` was merged and reverted.  ( #6225 and #6226 )

But I see no reason not use rest params (node 10 supports this), we already used it a lot.

- [ ] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory)
- [ ] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/pr-XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [x] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/master/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

**Comment by thorn0** [comment]
Do we really still need to transpile rest parameters? What for?

**Comment by fisker** [comment]
`bin-prettier.js` keeps `rest params syntax`, `standalone.js` still transpile it.

https://dev.azure.com/prettier/prettier/_build/results?buildId=4387&view=artifacts&type=publishedArtifacts

**Comment by thorn0** [comment]
But what for? For [obscure browsers](https://browserl.ist/?q=%3E0.25%25%2C+not+ie+11%2C+not+op_mini+all) like Safari 5 for Windows? Let's stop doing this, shall we? Something tells me Prettier doesn't work there anyway. We don't do any test in those browsers and don't really care.

**Comment by fisker** [comment]
How about https://github.com/eslint/eslint/blob/1b8a3ce15237b9085f2761dcf73655207e6169a6/webpack.config.js#L37

**Comment by thorn0** [comment]
`>0.5%, not ie 11, not safari 5.1, not op_mini all`

File: src/common/fast-path.js [hunk 1]
```diff
@@ -83,14 +86,17 @@ FastPath.prototype.callParent = function callParent(callback, count) {
 // accessing this.getValue()[name1][name2]... should be array-like. The
 // callback will be called with a reference to this path object for each
 // element of the array.
-FastPath.prototype.each = function each(callback /*, name1, name2, ... */) {
+FastPath.prototype.each = function each(
+  callback,
+  /* name1, name2, ... */ ...names
```
Comment (thorn0) [inline src/common/fast-path.js]: `/* name1, name2, ... */` - please remove such comments

**Comment by thorn0** [comment]
There are also `Array.prototype.push/unshift.apply` and `[].push.apply` calls in the code base. Can you rewrite them too while you're at it?

File: src/common/fast-path.js (hunk 1 above)
Comment (fisker) [reply src/common/fast-path.js]: Done

**Pull request merged**